In [1]:
import pandas as pd
import matplotlib.pyplot as plt
import seaborn as sns

In [2]:
# Set display options for better viewing
pd.set_option('display.max_columns', None)
pd.set_option('display.width', 200)

In [3]:
# Load the data
df = pd.read_csv('../data/netflix_shows_complete.csv')

In [4]:
# How many rows and columns?
print(f"Dataset shape: {df.shape}")
print(f"We have {df.shape[0]} shows and {df.shape[1]} features")

Dataset shape: (500, 23)
We have 500 shows and 23 features


In [5]:
# What columns do we have?
print(df.columns.tolist())

['id', 'name', 'first_air_date', 'popularity', 'vote_average', 'vote_count', 'status', 'in_production', 'num_seasons', 'num_episodes', 'genres', 'type', 'original_language', 'origin_country', 'avg_episode_runtime', 'show_age_days', 'days_since_last_episode', 'keywords', 'last_air_date', 'us_content_rating', 'imdb_id', 'created_by', 'homepage']


In [6]:
# First few rows
df.head()

id                           name first_air_date  popularity  vote_average  vote_count            status  in_production  num_seasons  num_episodes  \
0   66732                Stranger Things     2016-07-15    489.9853         8.600       20366             Ended          False            5            42   
1  308482   Taskaree: The Smuggler's Web     2026-01-14     99.6134         7.300           3  Returning Series           True            1             7   
2  259731                     HIS & HERS     2026-01-08     84.3181         7.006         154             Ended          False            1             6   
3  250505  Agatha Christie's Seven Dials     2026-01-15     79.0975         6.280          75             Ended          False            1             3   
4   63174                        Lucifer     2016-01-25     78.0905         8.437       15263             Ended          False            6            93   

                                          genres        type original_language origin_country  avg_episode_runtime  show_age_days  days_since_last_episode  \
0  Sci-Fi & Fantasy, Mystery, Action & Adventure    Scripted                en             US                  NaN         3477.0                     21.0   
1                          Crime, Mystery, Drama    Scripted                hi             IN                  NaN            7.0                      7.0   
2                          Drama, Crime, Mystery  Miniseries                en             US                  NaN           13.0                     13.0   
3                                 Drama, Mystery    Scripted                en             GB                  NaN            6.0                      6.0   
4                        Crime, Sci-Fi & Fantasy    Scripted                en             US                 53.5         3649.0                   1594.0   

                                            keywords last_air_date us_content_rating     imdb_id                created_by                                homepage  
0  monster, small town, indiana, usa, experiment,...    2025-12-31             TV-14   tt4574334  Ross Duffer, Matt Duffer  https://www.netflix.com/title/80057281  
1                                                NaN    2026-01-14             TV-MA  tt35335046             Neeraj Pandey  https://www.netflix.com/title/81406373  
2  loss of loved one, detective, based on novel o...    2026-01-08             TV-MA  tt33035373           William Oldroyd  https://www.netflix.com/title/81662954  
3     based on novel or book, clock, 1920s, thriller    2026-01-15             TV-14  tt31974288            Chris Chibnall  https://www.netflix.com/title/81314952  
4  based on comic, los angeles, california, devil...    2021-09-10             TV-14   tt4052886               Tom Kapinos  https://www.netflix.com/title/80057918

In [7]:
# Data types
df.dtypes

id                           int64
name                        object
first_air_date              object
popularity                 float64
vote_average               float64
vote_count                   int64
status                      object
in_production                 bool
num_seasons                  int64
num_episodes                 int64
genres                      object
type                        object
original_language           object
origin_country              object
avg_episode_runtime        float64
show_age_days              float64
days_since_last_episode    float64
keywords                    object
last_air_date               object
us_content_rating           object
imdb_id                     object
created_by                  object
homepage                    object
dtype: object

In [8]:
# Summary statistics
df.describe()

id  popularity  vote_average    vote_count  num_seasons  num_episodes  avg_episode_runtime  show_age_days  days_since_last_episode
count     500.000000  500.000000    500.000000    500.000000   500.000000    500.000000           195.000000     499.000000               498.000000
mean   150615.584000   12.228262      7.407288    794.402000     2.674000     42.440000            45.106154    1749.779559               994.333333
std     79825.958017   24.535885      0.845124   1996.612525     3.636781    186.365691            77.576530    1773.519833               907.244424
min       502.000000    4.441100      0.000000      0.000000     1.000000      1.000000             7.000000      -1.000000                 1.000000
25%     81826.500000    5.481950      7.000000     68.250000     1.000000      8.000000            25.000000     531.500000               247.250000
50%    119003.500000    7.295150      7.500000    212.000000     2.000000     16.000000            41.000000    1419.000000               712.500000
75%    230019.500000   10.974450      7.964500    681.750000     3.000000     32.000000            50.000000    2560.000000              1548.500000
max    310070.000000  489.985300     10.000000  20366.000000    56.000000   3161.000000          1100.000000   20526.000000              4191.000000

In [9]:
# Missing values
missing = df.isnull().sum()
missing[missing>0]

first_air_date               1
avg_episode_runtime        305
show_age_days                1
days_since_last_episode      2
keywords                    21
last_air_date                2
us_content_rating           38
imdb_id                      6
created_by                  89
homepage                     4
dtype: int64

In [10]:
# Status distribution
print(df['status'].value_counts())

status
Ended               294
Returning Series    138
Canceled             67
In Production         1
Name: count, dtype: int64


#### Initial Insights
- 67 shows have been cancelled, and 138 shows are returning.
- 13.4 % of shows have been canceled.
- Columns with missing data
    - first_air_date: 1 missing entry
    - avg_episode_runtime: 305 missing entries
    - show_age_days: 1 missing entry
    - days_since_last_episode: 2 missing entries
    - keywords: 21 missing entries
    - last_air_date: 2 missing entries
    - us_content_rating: 38 missing entries
    - imdb_id: 6 missing entries
    - created_by: 89 missing entries
    - homepage: 4 missing entries
- Minimum popularity rating: 4.4411
- Maximum popularity rating: 489.9853
